src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/SetAzureStorageAccountManagementPolicy.cs

```diff
@@ -0,0 +1,68 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Management.Storage.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Storage;
+using Microsoft.Azure.Management.Storage.Models;
+using System.Management.Automation;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Commands.Management.Storage
+{
+    [Cmdlet(VerbsCommon.Set, StorageAccountManagementPolicyNounStr, SupportsShouldProcess = true), OutputType(typeof(PSManagementPolicy))]
+    public class SetAzureStorageAccountManagementPolicyCommand : StorageAccountBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Storage Account Name.")]
+        [Alias(AccountNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountName { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Data Policy Rules.")]
+        [ValidateNotNullOrEmpty]
+        public object Policy { get; set; }
```

Comment:
Is there a more specific type for Policy than "Object"

Reply:
Since we need to input a JObject generete in Json 6 here, but AzureRM.Storage is reference Json10, so we can't specify the correct input object type Other than Object.